src/lib.rs

```diff
@@ -0,0 +1,420 @@
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+
+    /// Type of identifier for replies.
+    type ReplyId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Post {
+    title: Vec<u8>,
+    body: Vec<u8>,
+}
+
+impl Post {
+    fn new(title: Vec<u8>, body: Vec<u8>) -> Self {
+        Self { title, body }
+    }
+}
+
+// Blog`s pallet storage items.
+decl_storage! {
+    trait Store for Module<T: Trait> as BlogModule {
+
+        // Blog Ids set, associated with owner
+        BlogIds get(fn blog_ids_by_owner): map T::AccountId => Option<BTreeSet<T::BlogId>>;
+
+        // Post Ids set, associated with blog
+        BlogPostIds get(fn post_ids_by_blog_id): map T::BlogId => Option<BTreeSet<T::PostId>>;
+
+        // Root replies
+        BlogPostParentReplyIds get (fn post_root_reply_ids): map (T::BlogId, T::PostId) => Option<BTreeSet<T::ReplyId>>;
+
+        // Direct replies to reply
+        BlogPostChildReplyIds get (fn post_child_reply_ids): map (T::BlogId, T::PostId, T::ReplyId) => Option<BTreeSet<T::ReplyId>>;
+
+        BlogPost get(fn post_by_id): map (T::BlogId, T::PostId) => Option<Post>;
+
+        Reply get (fn reply_by_id): map (T::BlogId, T::PostId, T::ReplyId) => Option<Vec<u8>>;
+
+        BlogLockedStatus get(fn blog_locked): map T::BlogId => bool;
+
+        PostLockedStatus get(fn post_locked): map (T::BlogId, T::PostId) => bool;
+
+        //Reply Ids set, associated with owner
+        ReplyIds get (fn reply_ids_by_owner): map T::AccountId => Option<BTreeSet<(T::BlogId, T::PostId, T::ReplyId)>>;
+
+        BlogsCount get(fn blogs_count): T::BlogId;
+
+        // Overall post count by blog
+        PostsCount get(fn posts_count): map T::BlogId => T::PostId;
+
+        // Overall replies count by blog
+        RepliesCount get(fn replies_count): map (T::BlogId, T::PostId) => T::ReplyId;
+    }
+}
+
+// Blog`s pallet dispatchable functions.
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        // Initializing events
+        fn deposit_event() = default;
+
+        // Security/configuration constraints
+        const POST_TITLE_MAX_LENGTH: MaxLength = T::PostTitleMaxLength::get();
+        const POST_BODY_MAX_LENGTH: MaxLength = T::PostBodyMaxLength::get();
+        const REPLY_MAX_LENGTH: MaxLength = T::ReplyMaxLength::get();
+
+        const POSTS_MAX_NUMBER: MaxNumber = T::PostsMaxNumber::get();
+        const REPLIES_MAX_NUMBER: MaxNumber  = T::RepliesMaxNumber::get();
+        const DIRECT_REPLIES_MAX_NUMBER: MaxNumber = T::DirectRepliesMaxNumber::get();
+
+        const CONSECUTIVE_REPLIES_MAX_NUMBER: MaxConsecutiveRepliesNumber = T::ConsecutiveRepliesMaxNumber::get();
+
+        pub fn create_blog(origin) -> dispatch::Result {
```

Comment:
Blog creation is not something people can do on their own, its is a non-dispatchable `Module` method that can be invoked in the runtime only. No origin check is done. The spec said something slightly different

```
A blog is created via an extrinsic where access is gated by a dedicated EnsureOrigin runtime trait that returns no success value.
```

But lets ignore it for now.

Reply:
Ok

Reply:
Fixed https://github.com/iorveth/substrate-blog-module/commit/8a7c32befb198d3a0c73b7858b9aa2f03bf16fa7